src/MockGenerator/Program.cs

```diff
@@ -0,0 +1,372 @@
+        {
+            var baseFileName = Path.GetFileNameWithoutExtension(file);
+
+            fileDirectory = Path.GetDirectoryName(file);
+            Debug.Assert(fileDirectory != null, "fileDirectory != null");
+            return baseFileName;
+        }
+
+        private static InterfaceDeclarationSyntax DecorateInterfaceWithWrapperFunction(
+            MethodDeclarationSyntax methodDeclaration,
+            IdentifierNameSyntax invokeMethodIdentifier,
+            InterfaceDeclarationSyntax interfaceDeclaration)
+        {
+            var dllImport = methodDeclaration.AttributeLists
+                .First(x => x.OpenBracketToken.HasLeadingTrivia);
+            var interfaceMethodDeclaration = SyntaxFactory.MethodDeclaration(
+                SyntaxFactory.List<AttributeListSyntax>(),
+                SyntaxFactory.TokenList(),
+                methodDeclaration.ReturnType,
+                default(ExplicitInterfaceSpecifierSyntax),
+                invokeMethodIdentifier.Identifier,
+                methodDeclaration.TypeParameterList,
+                methodDeclaration.ParameterList,
+                methodDeclaration.ConstraintClauses,
+                default(BlockSyntax),
+                SyntaxFactory.Token(SyntaxKind.SemicolonToken));
+
+            interfaceDeclaration = interfaceDeclaration
+                .AddMembers(
+                    interfaceMethodDeclaration
+                        .WithTrailingTrivia(
+                            NewLineCharacter,
+                            TabCharacter)
+                        .WithLeadingTrivia(dllImport.OpenBracketToken.LeadingTrivia));
+            return interfaceDeclaration;
+        }
+
+        private static ClassDeclarationSyntax DecorateClassWithWrapperFunction(MethodDeclarationSyntax methodDeclaration,
+            IdentifierNameSyntax invokeMethodIdentifier,
+            ClassDeclarationSyntax classDeclaration)
+        {
+            var dllImport = methodDeclaration.AttributeLists
+                .First(x => x.OpenBracketToken.HasLeadingTrivia);
+            var arguments = methodDeclaration.ParameterList
+                .Parameters
+                .Select((x, i) =>
+                {
+                    var identifierName = SyntaxFactory.Argument(
+                        null,
+                        x.Modifiers.FirstOrDefault(z => z.IsKind(SyntaxKind.RefKeyword) || z.IsKind(SyntaxKind.OutKeyword)),
+                        SyntaxFactory.IdentifierName(x.Identifier));
+                    if (i > 0)
+                    {
+                        identifierName = identifierName.WithLeadingTrivia(WhitespaceCharacter);
+                    }
+
+                    return identifierName;
+                });
+
+            var arrowBody = SyntaxFactory.ArrowExpressionClause(
+                SyntaxFactory.Token(SyntaxKind.EqualsGreaterThanToken)
+                    .WithLeadingTrivia(
```

Comment:
I was speaking of whitespace trivia, not the documentation one.

The current code generate specific whitespace trivia (NewLike, TabCharacter) but if you use elastic trivia and run the Formatter it will generate the correct indentation code.

In essense when you generate code, it's better to avoid doing indentation & most whitespace handling by hand, the `Formatter` class is there for that.

It's only problem is that it depends on a configuration that isn't layered (As R# ones is) so it isn't saved in a commitable file and it's not extendable so the generated files might be fixed in a way stylecop doesn't like.
The first can be solved by changing the configuration of the workspace and the second one can be solved by running the solution-wide code fixe of any stylecop analyzer that disagree after the generation.